sklearn/neighbors/tests/test_dist_metrics.py

```diff
@@ -121,6 +121,29 @@ def haversine_slow(x1, x2):
                               np.sin(0.5 * D2) ** 2)
 
 
+def test_jensenshannon_metric():
+    def entropy(x):
+        return -np.nansum(x * np.log2(x))
+
+    def jensenshannon_slow(x1, x2):
+        mix = (x1 + x2) / 2
+        return np.sqrt(entropy(mix) - (entropy(x1) + entropy(x2)) / 2)
+
+    X = np.random.dirichlet([1]*10, 10)
```

Comment:
Here you are using the global np.random. You should never do this, and always define a local, seeded random number generator using 'rng = np.random.RandomState(0)'